### PR TITLE
Implemented land/lor/lxor tril tests

### PR DESCRIPTION
**Pull request by hsoontie**
Implemented land, lor and lxor tril tests for AArch64

### CODE REVIEW

**Comment by hsoontie** [comment]
Changes from master that were merged since my first version showed up by accident in my changes, will fix this soon

**Comment by hsoontie** [comment]
It should be good now, @Leonardo2718 !

File: fvtest/compilertriltest/LogicalTest.cpp [hunk 1]
```diff
@@ -103,3 +115,51 @@ INSTANTIATE_TEST_CASE_P(LogicalTest, Int32LogicalBinary, ::testing::Combine(
...
         std::make_tuple("ixor", ixor)
         )));
+
+class Int64LogicalBinary : public TRTest::BinaryOpTest<int64_t> {};
+
+TEST_P(Int64LogicalBinary, UsingConst) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[120] = {0};
+    std::snprintf(inputTrees, 120, "(method return=Int64 (block (lreturn (%s (lconst %li) (lconst %li)) )))", param.opcode.c_str(), param.lhs, param.rhs);
```
Comment (Leonardo2718) [inline fvtest/compilertriltest/LogicalTest.cpp]: In new tests, we've been trying to encourage splitting Tril code into multiple lines to make it easier to ready. Would you mind breaking up this string into multiple lines?

File: fvtest/compilertriltest/LogicalTest.cpp [hunk 2]
```diff
@@ -103,3 +115,51 @@ INSTANTIATE_TEST_CASE_P(LogicalTest, Int32LogicalBinary, ::testing::Combine(
...
+    volatile auto exp = param.oracle(param.lhs, param.rhs);
+    volatile auto act = entry_point();
+    ASSERT_EQ(exp, act);
+}
+
+TEST_P(Int64LogicalBinary, UsingLoadParam) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[120] = {0};
+    std::snprintf(inputTrees, 120, "(method return=Int64 args=[Int64, Int64] (block (lreturn (%s (lload parm=0) (lload parm=1)) )))", param.opcode.c_str());
```
Comment (Leonardo2718) [inline fvtest/compilertriltest/LogicalTest.cpp]: Same comment as above.

**Comment by hsoontie** [comment]
@Leonardo2718 Thank you! Edited again, let me know how it looks :)

File: fvtest/compilertriltest/LogicalTest.cpp [hunk 3]
```diff
@@ -103,3 +115,62 @@ INSTANTIATE_TEST_CASE_P(LogicalTest, Int32LogicalBinary, ::testing::Combine(
...
+
+TEST_P(Int64LogicalBinary, UsingConst) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[120] = {0};
+    std::snprintf(inputTrees, 120, "(method return=Int64 ("
+                                "block ("
+                                "lreturn ("
+                                "%s (lconst %li) (lconst %li)"
+                                ") )))",
```
Comment (Leonardo2718) [inline fvtest/compilertriltest/LogicalTest.cpp]: Not quite what I was thinking. :sweat_smile:

Here are some examples of how it's done in other files: 
https://github.com/eclipse/omr/blob/4d1af9870cec7db040cc9a785c8de9c16ae1e814/fvtest/compilertriltest/TypeConversionTest.cpp#L74-L78
https://github.com/eclipse/omr/blob/4d1af9870cec7db040cc9a785c8de9c16ae1e814/fvtest/compilertriltest/CompareTest.cpp#L58-L63

**Comment by hsoontie** [comment]
Edited again @Leonardo2718

**Comment by Leonardo2718** [comment]
@genie-omr build all

File: fvtest/compilertriltest/LogicalTest.cpp [hunk 4]
```diff
@@ -103,3 +115,66 @@ INSTANTIATE_TEST_CASE_P(LogicalTest, Int32LogicalBinary, ::testing::Combine(
         std::make_tuple("iand", iand),
         std::make_tuple("ixor", ixor)
         )));
+
+class Int64LogicalBinary : public TRTest::BinaryOpTest<int64_t> {};
+
+TEST_P(Int64LogicalBinary, UsingConst) {
+    auto param = TRTest::to_struct(GetParam());
+
+    char inputTrees[120] = {0};
```
Comment (Leonardo2718) [inline fvtest/compilertriltest/LogicalTest.cpp]: Hm... It looks like the length of the string needs to be increased. Maybe 160? You can try running the tests on your local machine to see if it works.

**Comment by Leonardo2718** [comment]
@genie-omr build all

**Comment by Leonardo2718** [comment]
Thanks @hsoontie for fixing that.

All checks have now passed. Merging.

**Pull request merged**